src/core/tsi/s2a/handshaker/s2a_handshaker_client.h

```diff
@@ -111,16 +111,47 @@ class S2AHandshakerClient {
    *  no further handshaker requests will be scheduled with the S2A. **/
   void Shutdown();
 
-  /** This method is exposed for testing purposes only. **/
-  grpc_byte_buffer* get_send_buffer_for_testing();
+  void MaybeCompleteTsiNext(
```

Comment:
Can we add some comments for it? especially why it is `Maybe`?

Reply:
I've added some comments in the next commit.

Reply:
I do not think the comment is accurate. 

If the handshaker client has a final result or failure, it checks if the status is received or not, and if it is received then it will invoke the TSI callback and if the status is not received, it will not invoke the callback. If the final result is not received, it will go ahead execute the callback.

Reply:
I've corrected this in the newest commit. Thanks! :)